sdk/cosmos/azure-cosmos/azure/cosmos/_execution_context/hybrid_search_aggregator.py

```diff
@@ -251,21 +253,45 @@ def _run_hybrid_search(self):
             self._format_final_results(drained_results)
             return
 
+        # Get the Components weight if any
+        if self._hybrid_search_query_info.get('componentWeights'):
+            component_weights = self._hybrid_search_query_info['componentWeights']
+            component_count = len(self._hybrid_search_query_info['componentQueryInfos'])
+            # If the number of weights is less than the number of components, we fill the rest with 1.0
+            # Unlikely to be the case since a bad request should be returned if weights are less than components,
+            # and any extra weights relative to components are ignored, but we handle it here anyway.
+            if len(component_weights) != component_count:
```

Comment:
Don't need this check, if the user specifies less or more weights than the number of components, backend will throw a syntax error that it is not a well formed query.

Client will not make any assumptions on this or try to manipulate it.